reposerver/cache/cache.go

```diff
@@ -139,11 +150,32 @@ func (c *Cache) GetGitReferences(repo string, references *[]*plumbing.Reference)
 }
 
 func manifestCacheKey(revision string, appSrc *appv1.ApplicationSource, namespace string, trackingMethod string, appLabelKey string, appName string, info ClusterRuntimeInfo) string {
+	trackingKey := trackingKey(appLabelKey, trackingMethod)
+	return fmt.Sprintf("mfst|%s|%s|%s|%s|%d", trackingKey, appName, revision, namespace, appSourceKey(appSrc)+clusterRuntimeInfoKey(info))
+}
+
+func trackingKey(appLabelKey string, trackingMethod string) string {
 	trackingKey := appLabelKey
 	if text.FirstNonEmpty(trackingMethod, string(argo.TrackingMethodLabel)) != string(argo.TrackingMethodLabel) {
 		trackingKey = trackingMethod + ":" + trackingKey
 	}
-	return fmt.Sprintf("mfst|%s|%s|%s|%s|%d", trackingKey, appName, revision, namespace, appSourceKey(appSrc)+clusterRuntimeInfoKey(info))
+	return trackingKey
+}
+
+// LogDebugManifestCacheKeyFields logs all the information included in a manifest cache key. It's intended to be run
+// before every manifest cache operation to help debug cache misses.
+func LogDebugManifestCacheKeyFields(message string, reason string, revision string, appSrc *appv1.ApplicationSource, clusterInfo ClusterRuntimeInfo, namespace string, trackingMethod string, appLabelKey string, appName string) {
+	if log.IsLevelEnabled(log.DebugLevel) {
```

Comment:
I don't think this check is required. 
Logrus already checks for this inside the `Debug` function:
https://github.com/sirupsen/logrus/blob/v1.8.1/logger.go#L195-L201

Reply:
It's just a small performance enhancement. Calculating the fields involves some JSON marshaling that I'd rather avoid unless we know we're going to log.